jenkins-plugin/pom.xml

```diff
@@ -47,7 +47,7 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.346.x</artifactId>
+        <artifactId>bom-2.361.x</artifactId>
```

Comment:
Clearer to merge #587 first.

Reply:
my initial plan was to merge https://github.com/jenkinsci/pipeline-maven-plugin/pull/587 first, but it needs more work. This PR does not change or rely on newer baseline